src/object/srv_obj_migrate.c

```diff
@@ -2640,6 +2640,33 @@ destroy_existing_obj(struct migrate_pool_tls *tls, unsigned int tgt_idx,
 		return rc;
 	}
 
+	if (tls->mpt_opc == RB_OP_REINT && unlikely(cont->sc_migrate_epoch < tls->mpt_max_eph)) {
+		struct ds_pool		*pool = cont->sc_pool->spc_pool;
+		struct pool_target	*target;
+		d_rank_t		 myrank;
+
+		crt_group_rank(NULL, &myrank);
+
+		ABT_rwlock_rdlock(pool->sp_lock);
+		rc = pool_map_find_target_by_rank_idx(pool->sp_map, myrank, tgt_idx, &target);
+		D_ASSERT(rc == 1);
+		ABT_rwlock_unlock(pool->sp_lock);
+
+		/* For healthy target, do not need to discard stale DTX entries. */
+		if (target->ta_comp.co_status != PO_COMP_ST_UPIN) {
+			rc = dtx_discard(cont, tls->mpt_max_eph);
```

Comment:
I suppose this is only local operation? (rename it to dtx_local_discard() ? )Otherwise we probably should ignore the failure somehow, since other targets will cancel it by itself?